tutorials/usage-modifying-logprob/index.qmd

```diff
@@ -0,0 +1,57 @@
+---
+title: Modifying the Log Probability
+engine: julia
+---
+
+```{julia}
+#| echo: false
+#| output: false
+using Pkg;
+Pkg.instantiate();
+```
+
+Turing accumulates log probabilities internally in an internal data structure that is accessible through the internal variable `__varinfo__` inside of the model definition.
+To avoid users having to deal with internal data structures, Turing provides the `Turing.@addlogprob!` macro which increases the accumulated log probability.
+For instance, this allows you to
+[include arbitrary terms in the likelihood](https://github.com/TuringLang/Turing.jl/issues/1332)
+
+```{julia}
+using Turing
+
+myloglikelihood(x, μ) = loglikelihood(Normal(μ, 1), x)
+
+@model function demo(x)
+    μ ~ Normal()
+    Turing.@addlogprob! myloglikelihood(x, μ)
+end
+```
+
+and to [reject samples](https://github.com/TuringLang/Turing.jl/issues/1328):
+
+```{julia}
+using Turing
+using LinearAlgebra
+
+@model function demo(x)
+    m ~ MvNormal(zero(x), I)
+    if dot(m, x) < 0
+        Turing.@addlogprob! -Inf
+        # Exit the model evaluation early
+        return nothing
+    end
```

Comment:
Also, by 'rejecting samples', can I check that this really means 'forcing the sampler to reject the samples'? In that we're relying on the implementation of the sampler doing this, rather than (e.g.) some AbstractMCMC code that checks the log probability and discards the sample if it's `-Inf`?

Reply:
That is indeed correct:)

Reply:
Gotcha! I'll add a few words to that effect, then merge.